[DOCS] Expands the analyzing phase with the analysis specific sub-phases (#1038)
elastic/stack-docs: docs/en/stack/ml/df-analytics/ml-dfa-phases.asciidoc

=== How a {dfanalytics-job} works

A {dfanalytics-job} is essentially a persistent {es} task. During its life 
cycle, it goes through four main phases:

* reindexing,
* loading data,
* analyzing,

>   {dfanalytics} that you run),

Author: ```suggestion
```

> * analyzing (the sub-phases of this phase are depends on the type of 

Reviewer: I think since you prefaced it with "main" you can omit the qualification here
```suggestion
* analyzing,
```

==== Loading data

After the reindexing is finished, the job fetches the needed data from the 
destination index. It converts the data into the format that the analysis 
process expects, then sends it to the analysis process.

[discrete]
==== Analyzing

In this phase, the job generates a {ml} model for analyzing the data. The 
specific phases of analysis vary depending on the type of {dfanalytics-job}.

{oldetection-cap} jobs have a single analysis phase called `computing_outliers`, 
in which they identify outliers in the data.

{regression-cap} and {classification} jobs have four analysis phases:

> . `feature_selection`: Identifies which fields are significant for the analysis. 

Reviewer 2: What you have is fine, but I think we can be a little more specific.
```suggestion
. `feature_selection`: Identifies which of the supplied fields are most relevant for predicting the target variable. 
```

. `coarse_parameter_search`: Identifies initial values for undefined 
  hyperparameters.
. `fine_tuning_parameters`: Identifies final values for undefined 

>   hyperparameters. See <<hyperparameter optimization>>.

Author: ```suggestion
  hyperparameters. See <<hyperparameters,hyperparameter optimization>>.
```

==== Loading data

After the reindexing is finished, the job fetches the needed data from the 
destination index. It converts the data into the format that the analysis 
process expects, then sends it to the analysis process.

[discrete]
==== Analyzing

In this phase, the job generates a {ml} model for analyzing the data. The 
specific phases of analysis vary depending on the type of {dfanalytics-job}.

{oldetection-cap} jobs have a single analysis phase called `computing_outliers`, 
in which they identify outliers in the data.

{regression-cap} and {classification} jobs have four analysis phases:

. `feature_selection`: Identifies which fields are significant for the analysis. 

> . `coarse_parameter_search`: The first part of 

Reviewer: If you want to make this more concise, perhaps something like this:
```suggestion
. `coarse_parameter_search`: Identifies initial values for undefined hyperparameters. 
```

  <<hyperparameters,hyperparameter optimization>> in which the process finds 
  suitable values for the hyperparameters that are required to run the analysis. 

> . `fine_tuning_parameters`: After the values of hyperparameters are narrowed 

Reviewer: Likewise, could possibly shorten like this:
```suggestion
. `fine_tuning_parameters`: Identifies final values for undefined hyperparameters. See <<hyperparameter optimization>>.
```

  down to a certain range in the `coarse_parameter_search` phase, the parameters 
  are further refined to an exact value in this second part of hyperparameter 
  optimization.

> . `final_training`: This is the phase where the {ml} model training takes place.

Reviewer: If you want to make these more concise, you can make them partial phrases. For example:
```suggestion
. `final_training`: Trains the {ml} model.
```

==== Loading data

After the reindexing is finished, the job fetches the needed data from the 
destination index. It converts the data into the format that the analysis 
process expects, then sends it to the analysis process.

[discrete]
==== Analyzing

When all the required data is loaded, the analyzing phase begins. The exact 
process always depends on the type of {dfanalytics} – for example, 
{classification} or {oldetection} – that is executed, you can find the 
description of these processes below ordered by analysis type. The analyzing 
phase is not reported in the `phase` property of the 
{ref}/get-dfanalytics-stats.html[Get {dfanalytics-jobs} statistics API], the 
specific sub-phases are reported instead.

> In this phase, the job generates a {ml} model for analyzing the dataset.

Reviewer: > In this phase, the job ....

I think this sentence would be a good introduction to this paragraph, instead of appearing at the end.

[discrete]

> ===== {oldetection-cap}

Reviewer: > ===== {oldetection-cap}

In my opinion, these subsections are unnecessary since it's quite a small amount of content in each.

> {oldetection-cap} has one sub-phase during analyzing that called 

Reviewer: IMO this can be combined into a single sentence.  For example:
```suggestion
{oldetection-cap} jobs have a single analysis phase called `computing_outliers`, in which they identify outliers in the data. 
```

`computing_outliers`. This is the phase when the {oldetection} analysis 
identifies outliers in the loaded data.

[discrete]

> ===== {regression-cap} and {classanalysis}

Reviewer: > ===== {regression-cap} and {classanalysis}

Per previous suggestion, I think this sub-section is unnecessary since it just contains a small list.

> {regression-cap} and {classification} share the same set of sub-phases. There 

Reviewer: If you accept my previous suggestions, I'd recommend merging this pair of sentences too. For example:
```suggestion
{regression-cap} and {classification} jobs have four analysis phases:
```

are four of them:

> * `feature_selection`: In this phase, the process iterates through the available 

Reviewer: If these phases occur in a specific order, it might make sense to use a numbered list here. I'd also suggest making each bullet point less verbose. Alternatively, you could remove the list and discuss each phase in order.

```suggestion
. `feature_selection`: Identifies which fields are significant for the analysis 
```

  field set to define which fields are significant for the analysis.

> * `coarse_parameter_search`: This is the first part of 

Reviewer 3: Descriptions here make sense to me. However, could we please also get @tveasey 's input?
